popper/util.py

```diff
@@ -48,30 +55,205 @@ def handler(signum, frame):
 
     return result
 
+def load_kbpath(kbpath, bias_filename):
+    if not bias_filename:
+        bias_filename = fix_path(kbpath, 'bias.pl')
+
+    with open(bias_filename, "r") as bias_file:
+        bias_string = bias_file.read()
+
+    return (fix_path(kbpath, "bk.pl"), fix_path(kbpath, "exs.pl"), bias_string)
+    
+def fix_path(kbpath, filename):
+    full_filename = os.path.join(kbpath, filename)
+    return full_filename.replace('\\', '\\\\') if os.name == 'nt' else full_filename
+
+def parse_settings():
+    args = parse_args()
+    
+    (bk_file, ex_file, bias_string) = load_kbpath(args.kbpath, args.bias_file)
+
+    return Settings(
+        bias_string,
```

Comment:
Ideally Settings would take strings for bias, examples and BK. This would make it a lot easier to just generate these programs in an experiment runner without having to create files. But pyswip makes loading a Prolog program from a string overly complicated. So the bias is a string and the BK and examples are files.

Taking these arguments separately (rather than as a folder path ala kbpath) makes it easier to load the files from different locations. In my experience, this can be convenient when running experiments which are minor variations on each other.